### ISSUE TITLE
Unexpected behavior when callback throws an error

### ISSUE THREAD

**Issue by erickoledadevrel** [google/google-api-nodejs-client#1159]
Consider the following small modification of the UrlShortener example in the README:

```js
urlshortener.url.get(params, (err, res) => {
  if (err) return console.error('The API returned an error: ' + err);
  console.log(`Long url is ${res.data.longUrl}`);
  throw new Error('This error is not from the API');
});
```

### Expected behavior

The final error is bubbled up the stack and logged in the console.

### Actual behavior

The final error is funneled back into my callback and treated as an API error:

```
Long url is https://www.google.com/
The API returned an error: Error: This error is not from the API
```

This seems to be a side effect of [setting the callback as the `catch()` handler for the promise](https://github.com/google/google-api-nodejs-client/blob/a783911c95a83fefb08e5fb49ee9817af5cbdab4/src/lib/apirequest.ts#L64). The catch executes not only for errors resulting from the API call but also for any errors thrown in the `then()` handler for a successful call.

I'm not sure what Promise interface Axios supports, but I think the solution would be to set the callback as the error handler for a specific `then()`, instead of as a `catch()`:

```js
createAPIRequestAsync<T>(parameters)
        .then(r => callback(null, r),
              e => callback(e));
```

**Comment by JustinBeckwith** [comment]
Thanks for calling this out!  Also double thanks for including the fix :) I submitted a PR over in #1161, so hopefully this clears things up!

**Comment by erickoledadevrel** [comment]
No problem, thanks for the quick PR!
